src/transformers/integrations/peft.py

```diff
@@ -127,28 +133,32 @@ def load_adapter(
         elif adapter_name in self.peft_config:
             raise ValueError(f"Adapter with name {adapter_name} already exists. Please use a different name.")
 
-        adapter_config_file = find_adapter_config_file(
-            peft_model_id,
-            revision=revision,
-            token=token,
-        )
-
-        if adapter_config_file is None:
-            raise ValueError(
-                f"adapter model file not found in {peft_model_id}. Make sure you are passing the correct path to the "
-                "adapter model."
+        if peft_config is None:
+            adapter_config_file = find_adapter_config_file(
+                peft_model_id,
+                revision=revision,
+                token=token,
             )
 
-        loaded_peft_config = PeftConfig.from_pretrained(
-            peft_model_id,
-            revision=revision,
-            use_auth_token=token,
-        )
+            if adapter_config_file is None:
+                raise ValueError(
+                    f"adapter model file not found in {peft_model_id}. Make sure you are passing the correct path to the "
+                    "adapter model."
+                )
+
+            peft_config = PeftConfig.from_pretrained(
+                peft_model_id,
+                revision=revision,
+                use_auth_token=token,
+            )
 
         # Create and add fresh new adapters into the model.
-        inject_adapter_in_model(loaded_peft_config, self, adapter_name)
+        inject_adapter_in_model(peft_config, self, adapter_name)
 
-        adapter_state_dict = load_peft_weights(peft_model_id, revision=revision, use_auth_token=token)
+        if not isinstance(peft_model_id, dict):
+            adapter_state_dict = load_peft_weights(peft_model_id, revision=revision, use_auth_token=token)
+        else:
+            adapter_state_dict = peft_model_id
```

Comment:
Just checking that we can' t add a new ` adapter_state_dict`  arg right?

Reply:
Why?

Reply:
Actually it turns out it is possible, please check [d05ef41](https://github.com/huggingface/transformers/pull/25721/commits/d05ef41d86c8f9566ac4977e752dd847e222daef)